### PR TITLE
feat(signup-codes): Signup codes everywhere, disable signup link in most places

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Our tests were not testing reality, tests still used signup links whereas in real life almost everyone sees signup codes. This changes that so signup codes are tested everywhere users would see them.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Builds on #3536 and #3532 and should be reviewed afterwards. @mozilla/fxa-devs - r?

File: packages/fxa-content-server/app/tests/spec/views/subscriptions_product_redirect.js [hunk 1]
```diff
@@ -72,16 +71,33 @@ describe('views/subscriptions_product_redirect', function() {
...
-      assert.isTrue(view.initializeFlowEvents.calledOnce);
-      assert.deepEqual(PaymentServer.navigateToPaymentServer.args, [
-        [
-          view,
-          config.subscriptions,
-          `products/${PRODUCT_ID}`,
-          { plan: 'plk_12345' },
-        ],
-      ]);
+      return render().then(() => {
```
Comment (vbudhram) [inline packages/fxa-content-server/app/tests/spec/views/subscriptions_product_redirect.js]: This seems unrelated to changes signup code changes?

File: packages/fxa-content-server/tests/functional/confirm.js [hunk 2]
```diff
@@ -54,63 +45,21 @@ registerSuite('confirm', {
...
+          .then(click(selectors.CONFIRM_SIGNUP_CODE.LINK_RESEND))
 
           // the test below depends on the speed of the email resent XHR
           // we have to wait until the resent request completes and the
           // success notification is visible
           .then(testSuccessWasShown())
       );
     },
-
-    'sign up with a restmail address, get the open restmail button': function() {
```
Comment (vbudhram) [inline packages/fxa-content-server/tests/functional/confirm.js]: I guess technically the `open email button` is no longer used by moving to codes? Maybe a follow up to remove the feature?

File: packages/fxa-content-server/tests/functional/connect_another_device.js [hunk 3]
```diff
@@ -143,639 +126,5 @@ registerSuite('connect_another_device', {
         )
         .then(noSuchElement(selectors.SMS_SEND.SUCCESS));
     },
-
-    'signup Fx Desktop, verify same browser': function() {
```
Comment (vbudhram) [inline packages/fxa-content-server/tests/functional/connect_another_device.js]: 💯

File: packages/fxa-content-server/tests/functional/oauth_sign_up.js [hunk 4]
```diff
@@ -13,33 +13,25 @@ const config = intern._config;
...
   openPage,
-  openVerificationLinkInDifferentBrowser,
-  openVerificationLinkInNewTab,
-  openVerificationLinkInSameTab,
-  switchToWindow,
   testElementExists,
-  testElementTextInclude,
   testUrlInclude,
-  visibleByQSA,
+  //  visibleByQSA,
```
Comment (vbudhram) [inline packages/fxa-content-server/tests/functional/oauth_sign_up.js]: Is this fixed in the separate PR to add email bouncing?

File: packages/fxa-content-server/tests/functional/sign_in_totp.js [hunk 5]
```diff
@@ -183,25 +184,33 @@ registerSuite('TOTP', {
...
     'can reset password, prompt for TOTP and login - same browser same tab': function() {
-      return this.remote
-        .then(confirmTotpCode(secret))
+      return (
+        this.remote
+          .then(confirmTotpCode(secret))
 
-        .then(click(selectors.SETTINGS.SIGNOUT, selectors.ENTER_EMAIL.HEADER))
+          .then(click(selectors.SETTINGS.SIGNOUT, selectors.ENTER_EMAIL.HEADER))
+          //.then(clearBrowserState())
```
Comment (vbudhram) [inline packages/fxa-content-server/tests/functional/sign_in_totp.js]: Is this still needed?

File: packages/fxa-content-server/app/scripts/views/confirm_signup_code.js [hunk 6]
```diff
@@ -72,8 +72,8 @@ class ConfirmSignupCodeView extends FormView {
     const options = {
       service: this.relier.get('service') || null,
     };
-    return account
```
[the remark this answers is not in the document]
Comment (shane-tomlinson) [reply packages/fxa-content-server/app/scripts/views/confirm_signup_code.js]: Yeah, the user object can notify other tabs of the signin and keep the tabs in Sync, account cannot.

File: packages/fxa-content-server/tests/functional/confirm.js (hunk 2 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/tests/functional/confirm.js]: @ryanfeeley and I were discussing this last week, a new link to open the email provider may end up on the confirm yer code page.

File: packages/fxa-content-server/tests/functional/oauth_sign_up.js (hunk 4 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/tests/functional/oauth_sign_up.js]: Ahha, yes, I can bring this back now!

File: packages/fxa-content-server/app/tests/spec/views/subscriptions_product_redirect.js (hunk 1 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/app/tests/spec/views/subscriptions_product_redirect.js]: It's needed to pass this change here: https://github.com/mozilla/fxa/pull/3477/files#diff-77c3a2279a222015eab66c6e7c1e8c9dL29

**Comment by shane-tomlinson** [comment]
> With codes now, I think we might be able to remove resume tokens?

Almost, almost! They are still used for both password reset and adding a secondary email. See https://github.com/mozilla/fxa/issues/2814

File: packages/fxa-content-server/tests/functional/confirm.js (hunk 2 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/tests/functional/confirm.js]: Opened https://github.com/mozilla/fxa/issues/3545

File: packages/fxa-content-server/app/scripts/views/mixins/signup-mixin.js [hunk 7]
```diff
@@ -32,12 +31,9 @@ export default {
 
         const options = {
           resume: this.getStringifiedResumeToken(account),
+          verificationMethod: 'email-otp',
```
[the remark this answers is not in the document]
Comment (shane-tomlinson) [reply packages/fxa-content-server/app/scripts/views/mixins/signup-mixin.js]: https://github.com/mozilla/fxa/issues/3561
